src/huggingface_hub/utils/_http.py

```diff
@@ -590,3 +591,41 @@ def _curlify(request: requests.PreparedRequest) -> str:
             flat_parts.append(quote(v))
 
     return " ".join(flat_parts)
+
+
+def _adjust_range_header(original_range: Optional[str], resume_size: int) -> Optional[str]:
+    """
+    Adjust HTTP Range header to account for resume position.
+    """
+    if not original_range:
+        return f"bytes={resume_size}-"
+
+    if "," in original_range:
+        warnings.warn(f"Multiple ranges detected - {original_range!r}, using full range after resume", UserWarning)
+        return f"bytes={resume_size}-"
+    # Regex to parse HTTP Range header
+    range_regex = re.compile(r"^\s*bytes\s*=\s*(\d*)\s*-\s*(\d*)\s*$", re.IGNORECASE)
+
+    match = range_regex.match(original_range)
+    if not match:
+        raise RuntimeError(f"Invalid range format - {original_range!r}.")
+    start, end = match.groups()
+
+    if not start:
+        if not end:
+            raise RuntimeError(f"Invalid range format - {original_range!r}.")
+
+        new_suffix = int(end) - resume_size
+        if new_suffix <= 0:
+            return None
```

Comment:
I feel that this condition should be "if resume_size is above the requested end, then fail".

```suggestion
        if int(end) < resume_size:
            raise RuntimeError(...)
```

Reply:
I think for this case, it means your download has been completed. So I think we dont have to let it fail, just exit the `http_get` function is okay.